### DISCUSSION TITLE
Q: writeJointBindMatricies

### DISCUSSION THREAD

**Discussion by hypnosnhendricks** [zeux/meshoptimizer#220] [Q&A]
In writeJointBindMatricies, the node_scale is calculated using the quantization scale divided by the the # of bits used for position compression, which seems to default to 14 bits.

Why does the bits used for compression pass affect the scale here? I expected it just to use the offset/scale calculated for the model and was confused as to why the bits are being taken into account here.

**Comment by zeux** [comment]
> Why does the bits used for compression pass affect the scale here? I expected it just to use the offset/scale calculated for the model and was confused as to why the bits are being taken into account here.

Right now gltfpack uses unnormalized integers to output positions. Since the only matrix (other than bone matrices) used to transformed skinned geometry is invbind, it needs to contain a dequant matrix, which converts from unnormalized integer to normalized integer by dividing by bit count, and then multiplies by the scale to get back into the original range.

**Reply by hypnosnhendricks** [reply]
I think I see. In writeVertexStream() you quantize the positions using the bits, which defaults to 14. I didn't realize gltfpack would quantize with a variable bit precision. What is the  reason for not using the full 16 bit precision when packing into the uint16 vector?

**Reply by zeux** [reply]
It's a quality vs transmission size tradeoff. Using fewer bits results in smaller size after compression.
